### DISCUSSION TITLE
Linear and Powerlaw strain hardening question?

### DISCUSSION THREAD

**Discussion by naveeniisc** [idaholab/moose#16449] [Q&A Modules: Solid mechanics] [answered]
Hi Moose Experts,

I'm new to the MOOSE environment and trying to run some Tensor Mechanics examples. I need to verify analytical solution of this paper "**Plastic Relaxation of the Transformation Strain Energy of a Misfitting Spherical Precipitate: Linear and Power-Law Strain Hardening**"  with moose numerical solution.  Can anyone suggest me where should I start? Is this sufficient to start with this examples **projects/moose/modules/tensor_mechanics/test/tests/recompute_radial_return/isotropic_plasticity_finite_strain.i  and projects/moose/modules/combined/test/tests/power_law_hardening/PowrLawHardening.i** ? Paper Attached in this discussion.  
[Earmme1981_Article_PlasticRelaxationOfTheTransfor.pdf](https://github.com/idaholab/moose/files/5670107/Earmme1981_Article_PlasticRelaxationOfTheTransfor.pdf)
 

Thanks for the assistance!
Naveen

**Comment by crswong888** [comment]
@hrishiv @albiedahal

**Comment by jiangwen84** [comment]
If you are new to MOOSE, I suggest you start to read the Tensor Mechanics documentation, i.e. https://mooseframework.inl.gov/source/materials/IsotropicPowerLawHardeningStressUpdate.html

**Reply by naveeniisc** [reply]
Thank you jiangwen84 let me go through this.

**Comment by naveeniisc** [comment]
Hi Moose Experts,

I could create the input file for the problem as mentioned in the last post (for isotropic plasticity

), however the solution is not converging. 

I have read this information here :https://mooseframework.inl.gov/modules/tensor_mechanics/Convergence.html

, but could not understand the process of setting up the convergence criteria? 

Can someone please explain the convergence criteria in detail.
 
Thanks in advanced,
Naveen     

[JKlee_test.txt](https://github.com/idaholab/moose/files/5924475/JKlee_test.txt)

**Reply by naveeniisc** [reply]
Hi @jiangwen84 @

can you explain my above post in detail?

Thanks in advance
naveen

**Reply by jiangwen84** [reply]
For mechanics problem, you have to use `Global Strain` system to setup strain periodicity. https://mooseframework.inl.gov/source/actions/GlobalStrainAction.html

Could you start with a simpler problem? e.g. 2D, without periodicity. Once successfully, add periodicity, and then 3D.

**Reply by naveeniisc** [reply]
sorry about late reply thank you @jiangwen84. i will follow your instruction.

**Comment by ngrilli** [comment, marked as answer]
Dear @naveeniisc 

I noticed you fixed the displacement only on one node, called boundary = 100

Can you try more standard boundary conditions for mechanics problems, 
like constraining your cubic RVE in a corner,
something like:

  [./z0_bot]
    type = DirichletBC
    variable = disp_z
    boundary = back
    value = 0.0
  [../]

  [./y0_bot]
    type = DirichletBC
    variable = disp_y
    boundary = bottom
    value = 0.0
  [../]

  [./x0_bot]
    type = DirichletBC
    variable = disp_x
    boundary = left
    value = 0.0
  [../]

Convergence problems are likely to appear if you don't constrain properly your system.
Best Regards,
Nicolò Grilli
National University of Singapore

**Reply by naveeniisc** [reply]
Dear @ngrilli  thanks for suggestion, let me try your solution.
